src/components/kytos/misc/StatusMenu.vue

```diff
@@ -0,0 +1,764 @@
+       * @returns {string} Returns a string that acts as the class for current HTML element.
+       * @description Sets color for current status table cell.
+       */
+      switch(true) {
+        case (!data.enabled):
+          return "darkWhite"
+        case (data.status_reason.includes("maintenance")):
+          return "yellow";
+        case (data.status == "UP"):
+          return "green";
+        case (data.status == "DOWN"):
+          return "red";
+      }
+    },
+    /**
+     * @param {Object} item - The normal table row object containing:
+     * @param {string} item.dpid - The data path id of the device the Object references.
+     * @param {string} item.id - The id of the device the Object references.
+     * @param {string} index - The index of the current table row Object. Usually id or name.
+     * @returns {string} Returns a string that acts as the icon for current HTML element.
+     * @description Sets the current "Hidden/Selected" table column button icon by verifying if Object id
+     *              is found within current list. Split for each individual table: Switches, Links, and Interfaces.
+     */
+    isStoredS(item, index) {
+      if (this.selectedSwitches[index]?.dpid == item.dpid) {
+        return "check-square"
+      } else {
+        return "square"
+      }
+    },
+    isStoredL(item, index) {
+      if (this.selectedLinks[index]?.id == item.id) {
+        return "check-square"
+      } else {
+        return "square"
+      }
+    },
+    isStoredI(item, index) {
+      if (this.selectedInterfaces[index]?.id == item.id) {
+        return "check-square"
+      } else {
+        return "square"
+      }
+    },
+    /**
+     * @param {string} data - The name of the selected data.
+     * @returns {string} Returns a string that acts as the class for current HTML element.
+     * @description Highlights selected data button. 
+     *              Split for each individual table: Switches, Links, and Interfaces.
+     */
+    selectedSwitchData(data) {
+      if (this.currentSwitchData == data) {
+        return "selectedButton"
+      }
+    },
+    selectedLinkData(data) {
+      if (this.currentLinkData == data) {
+        return "selectedButton"
+      }
+    },
+    selectedInterfaceData(data) {
+      if (this.currentInterfaceData == data) {
+        return "selectedButton"
+      }
+    },
+    show_infoPanel(s) {
+      /**
+       * @event showInfoPanel
+       * @param {Object} s - The switch data from the current row used to generate the info panel.
+       * @description Displays info panel when table row is clicked.
+       */
+      var content = {component: 'kytos-topology-k-info-panel-switch_info',
+                     content: this.switches[s],
+                     icon: "cog",
+                     title: "Switch Details",
+                     subtitle: this.switches[s].connection}
+      this.$kytos.$emit("showInfoPanel", content)
+    },
```

Comment:
Adding as a suggestion here some helper function to view/edit Link and Interface details, as you proposed for Switch. I think having those InfoPanel can be very helpful to help the network operator to deploy some changes while still on the Status Menu. Please let me know what you think.

```suggestion
    },
    show_infoPanelLink(l) {
      /**
       * @event showInfoPanel
       * @param {Object} l - The link data from the current row used to generate the info panel.
       * @description Displays info panel when table row is clicked.
       */
      var link = this.links[l]
      var subtitle = ""
      if(link.metadata.link_name !== undefined && link.metadata.link_name.length !== 0){
        subtitle = link.metadata.link_name
      }
      else{
        subtitle = link.id
        if(subtitle.length > 16) subtitle=subtitle.substr(0,16) + "..."
      }
      var content = {component: 'kytos-topology-k-info-panel-link_info',
                     content: link,
                     icon: "cog",
                     title: "Link Details",
                     subtitle: subtitle}
      this.$kytos.$emit("showInfoPanel", content)
    },
    show_infoPanelInterface(i) {
      /**
       * @event showInfoPanel
       * @param {Object} i - The interface data from the current row used to generate the info panel.
       * @description Displays info panel when table row is clicked.
       */
      var content = {component: 'k-interface-info',
                     content: this.interfaces[i],
                     icon: "cog",
                     title: "Interface Details",
                     subtitle: i}
      this.$kytos.$emit("showInfoPanel", content)
    },
```